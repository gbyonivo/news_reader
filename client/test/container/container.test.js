import Container from '../../src/container';
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import combinedReducers from '../../src/reducers';
import thunk from 'redux-thunk';

describe('container', () => {
  it('should render correctly' , () => {
    const store = applyMiddleware(thunk)(createStore)(combinedReducers);
    const actual = renderer.create(
      <Provider store={store}>
        <Container />
      </Provider>
    ).toJSON();
    expect(actual).toMatchSnapshot();
  })

  it('should render correctly' , () => {
    const store = applyMiddleware(thunk)(createStore)(combinedReducers);

    const wrapper = mount(
      <Provider store={store}>
        <Container />
      </Provider>
    );
    spyOn(store, 'dispatch').and.callThrough();
    store.dispatch({ type: 'LOAD_ARTICLES', payload: { articles: [] } });
    wrapper.update();
  
    const actual = store.dispatch.calls.first().args;
    const expected = [{
      payload: { articles: [] }, 
      type: 'LOAD_ARTICLES' }
    ];
  
    expect(actual).toEqual(expected);
  })
});