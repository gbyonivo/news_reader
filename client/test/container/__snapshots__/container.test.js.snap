// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container should render correctly 1`] = `
<div>
  <div
    className="articles"
  >
    <input
      className="articlesFilter"
      defaultValue=""
      onChange={[Function]}
      placeholder="Search Articles"
      type="text"
    />
  </div>
</div>
`;
